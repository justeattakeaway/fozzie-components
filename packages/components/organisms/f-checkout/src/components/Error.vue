<template>
    <card
        is-rounded
        has-outline
        is-page-content-wrapper
        card-heading-position="center"
        data-test-id="checkout-error-page-component"
        :class="[$style['c-checkout-error'], $style['c-checkout-error--verticalPadding']]"
    >
        <!-- TODO: Load image from CDN in future -->
        <sad-bag-icon-decorator
            data-test-id="checkout-error-page-image" />

        <h1
            :class="$style['c-checkout-error-heading']"
            data-test-id="checkout-error-page-heading">
            {{ $t('errorMessages.pageLoad.heading') }}
        </h1>

        <p
            :class="$style['c-checkout-error-description']"
            data-test-id="checkout-error-page-description">
            {{ $t('errorMessages.pageLoad.description') }}
        </p>
    </card>
</template>

<script>
import Card from '@justeat/f-card';
import SadBagIconDecorator from '../assets/images/jet-sad-bag.svg';
import '@justeat/f-card/dist/f-card.css';

export default {
    components: {
        Card,
        SadBagIconDecorator
    }
};
</script>

<style lang="scss" module>
.c-checkout-error {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;

    &.c-checkout-error--verticalPadding {
        padding: spacing(x8) spacing() spacing(x4);

        @include media('<=narrow') {
            border: none;
            padding-top: spacing(x2);
            padding-bottom: spacing(x2);
        }
    }

    .c-checkout-error-heading {
        @include font-size(heading-s);
        margin-top: spacing(x8);
        margin-bottom: 0;
    }

    .c-checkout-error-description {
        @include font-size(body-l);
        margin-top: spacing();
    }
}
</style>
