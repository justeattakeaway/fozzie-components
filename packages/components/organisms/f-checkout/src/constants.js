export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHODS = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];

export const TENANT_MAP = {
    'en-GB': 'uk',
    'en-AU': 'au',
    'en-NZ': 'nz',
    'da-DK': 'dk',
    'es-ES': 'es',
    'en-IE': 'ie',
    'it-IT': 'it',
    'nb-NO': 'no'
};

export const VALIDATIONS = {
    address: 'addressValidations',
    guest: 'guestValidations'
};
