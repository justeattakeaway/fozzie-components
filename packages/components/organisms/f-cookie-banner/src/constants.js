export const CONSENT_COOKIE_NAME = 'je-cookieConsent';
export const LEGACY_CONSENT_COOKIE_NAME = 'je-banner_cookie';
