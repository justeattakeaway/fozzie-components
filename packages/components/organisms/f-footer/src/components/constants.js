export const confianzaUrl = 'https://www.confianzaonline.es/empresas/justeatspain.htm';
export const confianzaScreenReaderText = 'Accede a Confianza Online y comprueba nuestro sello de calidad en Internet';
