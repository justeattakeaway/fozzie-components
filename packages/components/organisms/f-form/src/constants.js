export const FORM_EVENTS = {
    fieldUpdated: 'updated',
    submitting: 'form-submitting'
};

export const PROP_VALIDATION_MESSAGES = {
    requiredFormFields: '`formFields` Not found in `formData` prop, please add an array of `formFields`',
    invalidFormFields: '`formFields` are invalid, please add an array of formFields',
    requiredNameProperty: 'Each field in `formFields` should contain "name" property',
    requiredTranslationsProperty: 'Each field in `formFields` should contain "translations" property',
    requiredLabelProperty: 'Each field in `formFields` should contain "translations.label" property'
};

export const DEFAULT_BUTTON_TEXT = 'Submit';
