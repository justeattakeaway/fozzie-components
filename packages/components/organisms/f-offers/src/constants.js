export const POSTCODE_EMPTY = 'POSTCODE_EMPTY';
export const POSTCODE_INVALID = 'POSTCODE_INVALID';
