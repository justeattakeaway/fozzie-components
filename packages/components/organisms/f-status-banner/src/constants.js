/* eslint-disable import/prefer-default-export */
export const BASE_HERO_URL = 'https://d30v2pzvrfyzpo.cloudfront.net/status-banner/';
