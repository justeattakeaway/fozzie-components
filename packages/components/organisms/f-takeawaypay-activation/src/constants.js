export const ACTIVATION_STATE_NONE = 'none';
export const ACTIVATION_STATE_AVAILABLE_LOGGED_IN = 'available_logged_in';
export const ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN = 'available_not_logged_in';
export const ACTIVATION_STATE_SUCCEEDED = 'succeeded';
export const ACTIVATION_STATE_FAILED = 'failed';
