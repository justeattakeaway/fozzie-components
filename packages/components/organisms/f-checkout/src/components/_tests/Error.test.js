import { VueI18n } from '@justeat/f-globalisation';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Error from '../Error.vue';
import {
    i18n,
    $cookies,
    $logger,
    createStore,
    defaultCheckoutState
} from './helpers/setup';
import { CHEKOUT_ERROR_FORM_TYPE } from '../../constants';

const localVue = createLocalVue();

localVue.use(VueI18n);
localVue.use(Vuex);

describe('Error', () => {
    let wrapper;

    beforeEach(() => {
        // Arrange & Act

        wrapper = shallowMount(Error, {
            i18n,
            localVue,
            store: createStore({
                ...defaultCheckoutState,
                basket: {
                    id: 'MGFkMzgwZjgtYjY3Yi00Nz-v1',
                    total: 0
                }
            }),
            propsData: {
                errorFormType: CHEKOUT_ERROR_FORM_TYPE.default,
                redirectUrl: 'menu-jason-1'
            },
            mocks: {
                $cookies,
                $logger
            }
        });
    });

    afterEach(() => {
        jest.clearAllMocks();
    });


    it('should be defined', () => {
        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    it('should show the heading', () => {
        // Act
        const heading = wrapper.find('[data-test-id="checkout-error-page-heading"]');

        // Assert
        expect(heading.text()).toMatchSnapshot();
    });

    it('should show the description', () => {
        // Act
        const description = wrapper.find('[data-test-id="checkout-error-page-description"]');

        // Assert
        expect(description.text()).toMatchSnapshot();
    });

    describe('methods', () => {
        describe('`redirectToMenu` ::', () => {
            const oldWindowLocation = window.location;

            beforeAll(() => {
                delete window.location;

                window.location = Object.defineProperties(
                    {},
                    {
                        ...Object.getOwnPropertyDescriptors(oldWindowLocation),
                        assign: {
                            configurable: true,
                            value: jest.fn()
                        }
                    }
                );
            });

            afterAll(() => {
                window.location = oldWindowLocation;
            });

            describe('when invoked', () => {
                it('should redirect the customer back to the menu page so the user can navigate back from the error', () => {
                    // Act
                    wrapper.vm.redirectToMenu();

                    // Assert
                    expect(window.location.assign).toHaveBeenCalledWith('menu-jason-1');
                });

                describe('AND errorFormType is accessForbiddenError', () => {
                    beforeEach(async () => {
                        await wrapper.setProps({ errorFormType: CHEKOUT_ERROR_FORM_TYPE.accessForbidden });
                    });
                    describe('AND a menu basket cookie does not exist', () => {
                        it('should not delete the cookie', () => {
                            // Arrange
                            const cookieRemoveSpy = jest.spyOn(wrapper.vm.$cookies, 'remove');

                            // Act
                            wrapper.vm.redirectToMenu();

                            // Assert
                            expect(cookieRemoveSpy).not.toHaveBeenCalled();
                        });
                    });

                    describe('AND a menu basket cookie already exists', () => {
                        it('should delete the cookie', () => {
                            // Arrange
                            jest.spyOn(wrapper.vm.$cookies, 'get').mockReturnValue(() => 'je-mw-basket');
                            const cookieRemoveSpy = jest.spyOn(wrapper.vm.$cookies, 'remove');

                            // Act
                            wrapper.vm.redirectToMenu();

                            // Assert
                            expect(cookieRemoveSpy).toHaveBeenCalledWith('je-mw-basket-MGFkMzgwZjgtYjY3Yi00Nz-v1');
                        });
                    });
                });
            });
        });
    });
});
