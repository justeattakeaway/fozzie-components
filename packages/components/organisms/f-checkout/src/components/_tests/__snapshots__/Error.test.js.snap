// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should show the description 1`] = `"It’s a problem at our end, sorry. Your basket is safe and sound, though, so try again soon."`;

exports[`Error should show the heading 1`] = `"We can’t show you this page"`;
